File: 1431_kids_with_the_greatest_number_of_candies/solution.rb
# LeetCode 1431 - Kids With The Greatest Number Of Candies
# https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/

def kids_with_candies(candies, extra_candies)
  maximum = candies.max
  candies.map { |value| value + extra_candies >= maximum }
end

File: 1431_kids_with_the_greatest_number_of_candies/Solution.swift
// LeetCode 1431 - Kids With the Greatest Number of Candies
// https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/

class Solution {
    func kidsWithCandies(_ candies: [Int], _ extraCandies: Int) -> [Bool] {
        let maximum = candies.max()!
        return candies.map { $0 + extraCandies >= maximum }
    }
}
